pccs/setup.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+set -e
+
+mkdir -p /opt/intel/pccs/config/
+cd /opt/intel/pccs/config/
+
+if [ -n "$APIKEY" ]; then
+    apikey="$APIKEY"
+    sed -i "s/\"ApiKey\"[ ]*:[ ]*\"\",/\"ApiKey\": \"$apikey\",/" default.json
+else
+    echo "ERROR: You need to submit an APIKEY, otherwise your PCCS can not connect to the PCS"
+    exit 1
+fi
+
+if [ -n "$USERPASS" ]; then
+    userpass="$USERPASS"
+else
+    userpass="12345"
+fi
+userhash=$(echo -n "$userpass" | sha512sum | tr -d '[:space:]-')
+sed -i "s/\"UserTokenHash\"[ ]*:[ ]*\"\",/\"UserTokenHash\": \"$userhash\",/" default.json
+
+if [ -n "$ADMINPASS" ]; then
+    adminpass="$ADMINPASS"
+else
+    adminpass="54321"
+fi
+adminhash=$(echo -n "$adminpass" | sha512sum | tr -d '[:space:]-')
+sed -i "s/\"AdminTokenHash\"[ ]*:[ ]*\"\",/\"AdminTokenHash\": \"$adminhash\",/" default.json
```

Comment:
Are UserTokenHash and AdminTokenHash mandatory? What happens if they are left empty?

Reply:
Yes, it looks like both are mandatory, but just for the use of the retrieval tool. If they are empty, the retrieval tool always fails to fetch the data because the authentication fails. However, when attestating the MarbleRun coordinator it works. So i guess we can leave this empty, right? Are there possibly any further consequences?

(looks like passwords are not allowed to be left empty: https://github.com/intel/SGXDataCenterAttestationPrimitives/blob/7a732afa248c426bfaf373e0bb2ff8eec14bdccc/QuoteGeneration/pccs/install.sh)

Reply:
It depends on what happens if we leave them empty. Maybe you need to read code to find that out.
If empty passwords just make it impossible to connect, we should go with it.
If empty passwords allow to circumvent authentication, we should generate random passwords to lock everyone out.

Reply:
So the user token is only needed when fetching the data from the PCCS over the /platforms API. For all other data the user can retrieve, like /rootcacrl, there is apparently no authentication. The admin token is validated for other APIs where the admin can empty the cache, read all the data of the PCCS, change values of the PCCS or newly initialize the whole PCCS. 
It looks like it should be possible to circumvent both authentications. But at the moment I am still trying to figure out a correct curl command to do that.